src/NuGet.Clients/NuGet.PackageManagement.UI/Actions/UIActionEngine.cs

```diff
@@ -835,20 +835,17 @@ private PackageLicenseInfo GeneratePackageLicenseInfo(IPackageSearchMetadata met
             var results = new List<ResolvedAction>();
 
             Debug.Assert(userAction.PackageId != null, "Package id can never be null in a User action");
+
             if (userAction.Action == NuGetProjectActionType.Install)
             {
-                foreach (var target in targets)
-                {
-                    var actions = await _packageManager.PreviewInstallPackageAsync(
-                        target,
-                        new PackageIdentity(userAction.PackageId, userAction.Version),
-                        resolutionContext,
-                        projectContext,
-                        uiService.ActiveSources,
-                        null,
-                        token);
-                    results.AddRange(actions.Select(a => new ResolvedAction(target, a)));
-                }
+                results.AddRange(await _packageManager.PreviewProjectsInstallPackageAsync(
+                    targets,
+                    new PackageIdentity(userAction.PackageId, userAction.Version),
+                    resolutionContext,
```

Comment:
The else of this code here is about an *uninstall*, and in there the projects are evaluated sequentially. So given the implementation, I'm not sure uninstall would use the optimization. 

https://github.com/NuGet/Home/issues/9932 

Am I missing something?

Reply:
Create initial Resolved actions are looks still sequential. Since we have room for improvement I created another followup issue [#9956](https://github.com/NuGet/Home/issues/9956).  
For https://github.com/NuGet/Home/issues/9932  I only checked correctness part (restore happens only once per project) and added unit tests for it.

Reply:
So I found your answer somewhere else in the comments: 

> Here I am only testing correctness part, restore a once per project.
For improvement I created another follow up issue #9956 which will make create resolved actions parallel. Now execution of resolved action is parallel.

Which I imagine is caused by: 

https://github.com/NuGet/NuGet.Client/pull/3559/files#diff-1d985febe094810d0c99714587ee52dfR2169

There's a *slight* regression here because when the nominations come in, an actual restore will be run, but that'll be addressed by #9956

I will add more details in #9956